src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -293,12 +292,12 @@ class BlockImporter(
     }
   }
 
-  private def importWith(importFuture: => Future[NewBehavior])(state: ImporterState): Unit = {
+  private def importWith(importTask: Task[NewBehavior])(state: ImporterState): Unit = {
     context become running(state.importingBlocks())
-    importFuture.onComplete {
-      case Failure(ex) => throw ex
-      case Success(behavior) => self ! ImportDone(behavior)
-    }
+    importTask
+      .map(self ! ImportDone(_))
+      .onErrorHandle(ex => log.error(ex, ex.getMessage))
```

Comment:
I am not sure here. Maybe we need to throw the exception?
What do you think?
.onErrorHandle(ex => {
  log.error(ex, ex.getMessage)
  Task.raiseError(ex)}
)

Reply:
The result is discarded with `runAsyncAndForget` so any exception will just be lost. The previous code wasn't handling the exception either because it was throwing it inside an `onComplete` callback.